.github/workflows/gradle.yml

```diff
@@ -32,10 +32,13 @@ jobs:
           java-version: ${{ matrix.Java }}
           check-latest: true
 
-      - name: Build and Test
-        run: |
-          ./gradlew --version
-          ./gradlew test --info
+      - uses: burrunan/gradle-cache-action@v1
+        # See https://github.com/burrunan/gradle-cache-action
+        name: Build and Test
+        with:
+          # It allows different cache contents for different JDKs
+          job-id: java${{ matrix.java }}
+          arguments: --scan --no-parallel --no-daemon check
```

Comment:
`--scan` is, well, for Gradle Build Scan (simplified UI for issue analysis)
`--no-parallel`, `-no-daemon`. Well, they are not strictly needed, however, I tend to copy-paste them from project to project. GitHub Actions does not provide many CPU, so it is unclear if going parallel would help.

Reply:
If the arguments are unrelated to caching, I'd rather go without them in this PR and think about adding them later.

Reply:
Well, `--no-daemon` is still helpful since we execute Gradle just once, and it makes no sense to keep the daemon after execution. Terminating the daemon after the build would save the resources for subsequent CI steps.